elements/pl-matching/pl-matching.py

```diff
@@ -0,0 +1,440 @@
+import prairielearn as pl
+import lxml.html
+import random
+import math
+import chevron
+
+
+WEIGHT_DEFAULT = 1
+FIXED_STATEMENTS_ORDER_DEFAULT = False
+FIXED_OPTIONS_ORDER_DEFAULT = False
+INLINE_DEFAULT = False
+PARTIAL_CREDIT_DEFAULT = True
+HIDE_ANSWER_PANEL_DEFAULT = False
+HIDE_HELP_TEXT_DEFAULT = False
+DETAILED_HELP_TEXT_DEFAULT = False
+HIDE_SCORE_BADGE_DEFAULT = False
+BLANK_DEFAULT = False
+BLANK_ANSWER = ' '
+NOTA_DEFAULT = False
+COUNTER_TYPE_DEFAULT = 'lower-alpha'
+
+
+def get_form_name(answers_name, index):
+    return f'{answers_name}-dropdown-{index}'
+
+
+def alpha_counter(i, letter_a):
+    """Converts a base-10 integer to base 26, using letter_a as the "0" digit."""
+    result = ''
+    while i > 0:
+        i -= 1
+        result = chr(ord(letter_a) + i % 26) + result
+        i //= 26
+
+    return result
+
+
+def get_counter(i, counter_type):
+    """Converts an integer counter to the specified CSS counter type"""
+    if counter_type == 'lower-alpha':
+        return alpha_counter(i, 'a')
+    elif counter_type == 'upper-alpha':
+        return alpha_counter(i, 'A')
+    elif counter_type == 'decimal':
+        return str(i)
+    else:
+        raise Exception('Illegal counter-type in pl-matching element: "' + counter_type + '" should be "decimal", "lower-alpha" or "upper-alpha".')
+
+
+def legal_answer(answer, counter_type, options):
+    """Checks that the given answer is within the range of the given counter type."""
+    if counter_type == 'lower-alpha':
+        return 'a' <= answer <= chr(ord('a') + len(options))
+    elif counter_type == 'upper-alpha':
+        return 'A' <= answer <= chr(ord('A') + len(options))
```

Comment:
These don't work with `aa`, `ab`, etc. I'm wondering if it isn't easier to have a check if the answer is in the valid keys? In other words, if there is a `pl-option` with that key?

Reply:
This would imply changing the options in params to use the converted key (e.g., `a`) instead of the index.